### DISCUSSION TITLE
Get referenced pull request from issue

### DISCUSSION THREAD

**Discussion by vetemi** [community/community#24367] [Apps, API and Webhooks]
<p>Hey Community,</p>
<p>Github allows to see if pull requests reference an issue (or vice versa), e.g. see the following example (randomly picked):</p>
<p><a href="https://github.com/bigdatagenomics/adam/issues/479" rel="nofollow noopener">issue with referenced pull request</a></p>
<p>This is great and I’m wondering if I can achieve the same via the API. While the API (v3) allows to see if commits are referenced in the issue (via its events) I wasn’t able to find a way to get referenced pull requests. I went through a couple of sources (e.g. <a href="https://help.github.com/en/articles/searching-issues-and-pull-requests" rel="nofollow noopener">here</a>).</p>
<p>Is this even possible?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You’re correct. I wasn’t able to find a way to do what you’re asking via the REST API. But here’s a query that works for the GraphQL API:</p>
<pre><code>query {
  repository(owner: "bigdatagenomics", name: "adam") {
    issue(number: 479) {
      timelineItems(first: 100) {
        nodes {
          ... on ClosedEvent {
            closer {
              ... on PullRequest {
                number
                title
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>which results in:</p>
<pre><code>{
  "data": {
    "repository": {
      "issue": {
        "timelineItems": {
          "nodes": [
            {},
            {},
            {},
            {
              "closer": {
                "number": 570,
                "title": "A few small conversion fixes"
              }
            }
          ]
        }
      }
    }
  }
}
</code></pre>
<p>I hope that helps!</p>

**Comment by vetemi** [comment]
<p>Thanks for your answer! Based on your query I played around with the v4 API and came up with something different which suits me better:</p>
<pre><code>query {
  repository(owner: "bigdatagenomics", name: "adam") {
    issue(number: 479) {
      timelineItems(first: 100) {
        nodes {
          ... on CrossReferencedEvent {
            source {
              ... on PullRequest {
                number
                title
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>In this case I get every referenced Pull Requests and not only those which closed the issue.</p>
<p>Thanks again!</p>

**Comment by lee-dohm** [comment]
<p>Awesome! Thanks for sharing your ultimate solution back :grinning:</p>

**Comment by elanzini** [comment]
<p>Hi, I am adding to this thread since the problem I am facing is similar.<br>
I am trying to write a query to retrieve the Pull Requests mentioned in one Issue (Note: Not the ones linked to the issue).</p>
<p>For example, looking at this issue <a href="https://github.com/ARMmbed/mbed-os/issues/11803" rel="noopener nofollow ugc">#11803</a>, there is a PR mentioned in <a href="https://github.com/ARMmbed/mbed-os/issues/11803#issuecomment-608270561" rel="noopener nofollow ugc">this comment</a>.</p>
<p><code>CrossReferencedEvent</code> does not include the PRs mentioned in this case.</p>
<pre><code class="lang-auto">query {
  repository(owner: "ARMmbed", name: "mbed-os") {
    issue(number: 11803) {
      timelineItems(first: 100) {
        nodes {
          ... on CrossReferencedEvent {
	          source {
              ... on PullRequest {
                number
                title
              }
          	}
        	}
      	}
    	}
  	}
  }
}
</code></pre>
<p>Result:</p>
<pre><code class="lang-auto">{
  "data": {
    "repository": {
      "issue": {
        "timelineItems": {
          "nodes": [
            {},
            {
              "source": {}
            },
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {
              "source": {}
            }
          ]
        }
      }
    }
  }
}
</code></pre>
<p>I have also tried exploring the different <code>IssueTimelineItems</code> but I could not find any that solved the problem.</p>
<p>This also does not solve it:</p>
<pre><code class="lang-auto">query {
  repository(owner: "ARMmbed", name: "mbed-os") {
    issue(number: 11803) {
      timelineItems(first: 100) {
        nodes {
          ... on ReferencedEvent {
             subject{
              ... on PullRequest {
                title
                number
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by KastanDay** [comment]
I wrote an elegant solution in Python as well: https://github.com/orgs/community/discussions/24492#discussioncomment-7469561
